src/main/native/org/apache/commons/crypto/random/OpenSslCryptoRandomNative.c

```diff
@@ -294,49 +280,57 @@ static unsigned long pthreads_thread_id(void)
  * If using an Intel chipset with RDRAND, the high-performance hardware
  * random number generator will be used.
  */
-static ENGINE * openssl_rand_init(void)
+static ENGINE * openssl_rand_init(JNIEnv *env)
 {
-  locks_setup();
+  if (dlsym_OpenSSL_version_num() < VERSION_1_1_X) {
+    locks_setup(env);
+	static void (*dlsym_ENGINE_load_rdrand) (void);
+	dlsym_ENGINE_load_rdrand = do_dlsym(env, openssl, "ENGINE_load_rdrand");
+	dlsym_ENGINE_load_rdrand();
+  }
 
-  dlsym_ENGINE_load_rdrand();
   ENGINE *eng = dlsym_ENGINE_by_id("rdrand");
 
-  int ret = -1;
-  do {
-    if (NULL == eng) {
-      break;
-    }
+    int ret = -1;
```

Comment:
nit: mind undoing these changes? Previous indentation was correct.

(Then we can see what's actually changing.)